add feedback

r03ert0 opened the pull request:
add emoji feedback to the room.html

r03ert0 committed:
add feedback

r03ert0 commented:
example:
![robot](https://user-images.githubusercontent.com/2310732/85932075-560f4500-b8c9-11ea-8397-ec6412af93e8.gif)

yarikoptic commented:
Yet to check - would it interfere with the chat if opened? 

If does- adding collapsing button or smth like that could mitigate

yarikoptic commented:
May be expand that panel with urls/icons for brainweb and gather town, and to the PDF if is known for the poster?

yarikoptic commented:
I don't get it - how bunnies get reflected in the robots glass clock? ;-)

yarikoptic commented:
Hey @all-contributors please add @r03ert0 for code design

yarikoptic commented on room.html:
```diff
@@ -2,11 +2,21 @@
     <div id="meet"/>
+
+    <div id="extras">
+        <div class="tool" onclick="feedback('🐇')">🐇</div>
```
for me (local deployment, running `/usr/bin/python3 -m http.server 8081 `) all those show up as
![DeepinScreenshot_select-area_20200628123704](https://user-images.githubusercontent.com/39889/85953100-1a619300-b93c-11ea-8fcd-6f39bfde5c34.png)

r03ert0 replied:
For the icons not showing up in local development, maybe copy/pasting the emojis could be replaced by explicitly typing their unicode codes? like `console.log("\u{1F430}")` (I used this handy website for the conversion: https://r12a.github.io/app-conversion/)

r03ert0 replied:
```html
<div class="tool" onclick="feedback(`${"\u{1F430}"}`)">&#x1F430;</div>
```

r03ert0 replied:
what's your browser? does it display the icons in the brain-web jisti room here? https://brain-web.github.io/videoconference/

yarikoptic replied:
Chromium
Version 79.0.3945.130 (Developer Build) built on Debian bullseye/sid, running on Debian bullseye/sid (64-bit)

https://brain-web.github.io/videoconference/ works ok, it is only when I tried this branch locally (may be should try brain-web. locally?)

yarikoptic replied:
I wonder if it just needs some html or css setup for those to appear correctly, and which was not copied over from brain-web?

yarikoptic replied:
e.g. they (obscure characters) are black not white like on brain-web

yarikoptic replied:
BTW 
- do they work for you when you test them locally on your laptop? (just try `/usr/bin/python3 -m http.server 8081` and go to localhost:8081 ;))
- they appear incorrect in "inspect" view for this page, but correctly in the one for brainweb

yarikoptic replied:
also i see
```
Uncaught ReferenceError: api is not defined
    at feedback.js:29
```
in errors within inspect -- I guess smth to fix

r03ert0 replied:
ha! funny... with the python server I get the currency icons instead of the emojis, like you
(normally I use the `http-server` I installed with `npm`)

r03ert0 replied:
the python server wanted to be told specifically the encoding to use. Adding this to room.html solved the problem:

```html
<meta content="text/html;charset=utf-8" http-equiv="Content-Type">
<meta content="utf-8" http-equiv="encoding">
```

r03ert0 replied:
I also made a function to install the event listener for the data channel instead of having wild code in feedback.js

yarikoptic reviewed (changes requested):
Need to get icons working for greedy myself on a local deployment.

There is indeed occlusion of chat by icons but I think it is tolerable and could be worked around by resizing window so chat text moves and/or scrollbar appears.

r03ert0 commented:
Ok! I'll try to add the button for PDF download and gather.town. Right now the info from the main page is sent to the room page via the url. Adding the poster's url to the parameters would make for a very messy address... i'll try using localStorage.

For the icons occluding the chat, and alternative would be to displace the whole jitsi div element to the right, and put all 'custom' tools to the left. Should I try that?

yarikoptic commented:
> For the icons occluding the chat, and alternative would be to displace the whole jitsi div element to the right, and put all 'custom' tools to the left. Should I try that?

probably not worth it... let's just have icons fixed up and/or added those new ones and merge to proceed.

yarikoptic commented:
Hey @r03ert0 -- will you have time/interest to tune it up?  I still like the idea of the feedback icons! ;)

r03ert0 commented:
yes! sorry. It was our 1st day back at the institute...

r03ert0 committed:
fix encoding issue

r03ert0 committed:
fix undefined `api` issue

yarikoptic commented:
cool.  Testing locally again -- It works BUT
- they are black (not those nice one pictures you showed) and thus not visible on black at all but fly though the screen black when clicked
- I guess in vue redesign (if I merge current gh-pages) - it is a different .html or place where to embed this too since it is not there at all.

soichih commented:
You can add it to `/ui/src/components/Room.vue`.. and icons / assets can be added to `/ui/src/assets/img`. From the vue template, you can reference it like this

```
<img src="@/assets/img/datalad_logo.svg" align="right" width="75px">
```

To test it locally, you can run `npm run dev` and it should launch a web browser. `npm run install` first to install all the dependencies.

r03ert0 commented:
the emojis are not icons, they are just unicode characters (for adding more emojis, you just copy/paste them into the html). Maybe what happens is that the default font in chromium does not have the emoji characters?

yarikoptic commented:
> the emojis are not icons, they are just unicode characters (for adding more emojis, you just copy/paste them into the html). Maybe what happens is that the default font in chromium does not have the emoji characters?

yes, could be -- or just the font you list is N/A? which should it be? or you don't assign any specific?

r03ert0 commented:
I don't assign any specific font... what's weird is that you do get the brain-web icons showing up normally, right?

yarikoptic commented:
yeah - may be my setup deficiency. But could you assign them white color so they are at least visible for me?  or that would ruin them for you?

yarikoptic commented:
FWIW -- I quickly verified on https://brain-web.github.io/ohbm2020/room.html#1155.ohbm2020-1155 that changing color to white for `tool` makes them visible in the panel.  They remain flying black... looking into installing more fonts with the hope to get better ones, although I do have e.g. fonts-noto-color-emoji pkg installed, but not sure what fonts chromium picks up

yarikoptic commented:
oh, have a look https://github.com/jgm/pandoc/issues/5469 

I learned from a reader today of a obscure & delightful iOS feature.

iOS (but apparently no other OS*) chooses to render a number of Unicode entities, including 'leftwards_arrow_with_hook', as big colorful emoji. This applies to all web pages, including ones which did not ask for it in any way. (This might also screw with any CSS modifications of footnotes one might do, but I'm not sure about it.)

yarikoptic commented:
so if we would like for this feature to work for others and be as pretty -- let's add/use icons?

r03ert0 commented:
ok! I guess we can download the emoji pngs from emojipedia and replace the relevant parts of the code to `<img>` tags. I'll try that!

r03ert0 committed:
change emojis for pngs

r03ert0 commented:
done! I changed the emojis for pngs downloaded from emojipedia. In `room.html` I replaced the unicode characters for `<img />` tags, and updated the css. I also updated `feedback.js` to use `img`.
If there were a way in the future to go back to just copy pasting the emoji unicodes, that would make the code easier to customise. However, downloading pngs is not that difficult either :P
Let me know if it works.

yarikoptic commented on assets/js/feedback.js:
```diff
@@ -14,7 +14,22 @@ function feedbackAnimationStep(timer, elem) {
+      elem.innerHTML = `<img class="emoji" src="/assets/images/emoji_brain_1f9e0.png" />`;
+      break;
+    case 'heart':
+        elem.innerHTML = `<img class="emoji" src="/assets/images/emoji_brain_1f9e0.png" />`;
```
needs to be a heart

yarikoptic commented on assets/js/feedback.js:
```diff
@@ -14,7 +14,22 @@ function feedbackAnimationStep(timer, elem) {
+        elem.innerHTML = `<img class="emoji" src="/assets/images/emoji_brain_1f9e0.png" />`;
+      break;
+    case 'rocket':
+        elem.innerHTML = `<img class="emoji" src="/assets/images/emoji_smiling-face-with-smiling-eyes-and-three-hearts_1f970.png" />`;
```
needs to be a rocket?

yarikoptic reviewed (changes requested):
IT WORKS! But some emjois are incorrectly "assigned"

r03ert0 committed:
fix mis-assigned emojis

r03ert0 commented:
oopsy! (can't code at 4am anymore 😅 )

yarikoptic commented:
Ok, so now master should me merged in, and this added to the new platform as well?

Also still would be nice to get PDF icon IMHO

yarikoptic commented:
@r03ert0 : We just merged @soichih 's #163 which added etherpad integration.  Would be nice to have similar (to "Notes") "feedback" button (on the right) which would provide this suggested pane with emojis!

yarikoptic commented:
Now that OHBM is over, let's work a bit more to move GallOP into an independent project.  I will keep it open, so that if this PR is finalized before #169 is closed, it could become a part of the "move". Otherwise -- it would make sense to close it then and I hope that you @r03ert0 would refile it into the new GallOP repository - getting nice visual feedback would be a nice feature to add.
Cheers!

r03ert0 commented:
sure : )
